src/tasks/MonoTargetsTasks/MarshalingPInvokeScanner/MarshalingPInvokeScanner.cs

```diff
@@ -0,0 +1,333 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Reflection;
+using System.Reflection.PortableExecutable;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+internal enum Compatibility
+{
+    Compatible,
+    Incompatible,
+    Inconclusive
```

Comment:
I think this could use a summary comment.  Particularly something about what types might be inconclusive in the first path.

Something like:

> For some valuetypes we cannot determine if they are compatible with disabled runtime marshaling without first resolving their base types.
> In this case we first mark the assembly as Inconclusive and do a second pass over the collected base type references in order to decide.
> If the base types are `System.Enum`, then the valuetypes are enumerations, and are compatible.